libqtile/layout/xmonad.py

```diff
@@ -237,16 +237,16 @@ def cmd_normalize(self, redraw=True):
             self.group.layout_all()
         self.do_normalize = False
 
-    def cmd_reset(self, redraw=True):
+    def cmd_reset(self, ratio=None, redraw=True):
         "Reset Layout."
-        self.ratio = self._med_ratio
+        self.ratio = ratio or self.default_ratio
         if self.align == self._right:
             self.align = self._left
         self.cmd_normalize(redraw)
 
     def _maximize_main(self):
         "Toggle the main pane between min and max size"
-        if self.ratio <= self._med_ratio:
+        if self.ratio <= self.default_ratio:
```

Comment:
Wouldn't it make sense for this number just to be halfway between `max_ratio` and `min_ratio` and not linked to `default_ratio` at all?

Reply:
I think this way makes sense, if I have a "fresh layout" and use `_maximize_main` I expect it to always maximize the master (because of the way it's named).

But if I have say `ratio = 0.6`, `max_ratio = 0.85` and `min_ratio = 0.15` in my config it would actually minimize since the half-way threshold is 0.5.

Reply:
Well, if you're going by the name then it should never set it to the min ratio...

I don't feel particularly strongly about it so if you think it's best as you've got it, I won't argue!

Reply:
Hmm you are right, I got cought up in the name and forgot it is a "toggle point" between what's considered "maximizable" and "minimizable". It does make sense to have `default_ratio` to be the toggle point, but it also makes sense to just have it in the middle.

I think I'll change it to the midpoint like you suggested, and the user can always inherit from the layout and modify the method (as it is a very short one).